Plz merge my commits into feature/db from feature/icon-recommendation

Mango-Juice opened the pull request:
Part 5에 속하는 feature/db 브랜치와 feature/icon-recommendation 브랜치를 합칩니다.

Mango-Juice committed:
docs : feature/db 브랜치 설명 추가

README.md 파일에 db 브랜치에 대한 설명을 추가함

CaChiJ committed:
docs : main-dial 브랜치 설명 추가

readme 문서에 feature/dir-main-page/main-dial 브랜치에 대한 설명을 추가함.

Mango-Juice committed:
feat : 아이콘 추천 기능 추가

Json 형태의 벡터 정보 파일을 가져올 수 있도록 구현함.
벡터 정보를 바탕으로 어떤 문자열과 유사도가 가장 높은 아이콘의 파일명을 구할 수 있도록 함.

Mango-Juice committed:
fix : 아이콘 추천시 메모리 초과 오류 및 응답 없음 현상 해결

아이콘 추천을 위한 벡터 정보를 담는 파일을 json에서 csv로 변경하여 시간/공간 복잡도를 개선함.
이를 통해 기존에 발생하던 메모리 초과 오류 및 응답 없음 현상을 일부 해결함.
다만 긴 로딩 시간과 큰 메모리 소모 문제를 완전히 해결한 것은 아님.

CaChiJ commented on app/src/main/java/com/example/plandial/IconRecommendation.java:
```diff
@@ -0,0 +1,154 @@
+        // 최초 1회 -> 파일 준비
+        if (!sIsReady) {
+            if(getIconVectors(context) && getWordVectors(context)){
+                sIsReady = true;
```
sIsReady = !sIsReady && getIconVectors(context) && getWordVectors(context) 로 바꾸셔도 됩니다.

Mango-Juice replied:
변경 완료하였습니다. ([refactor : 폴더명 변경, 코드 간략화 및 상수 선언](https://github.com/ProjectFRIDAY/Plan-Dial/pull/4/commits/6054c425cedadc63f1f2c03ad5565168f4fec0da))

CaChiJ commented on app/src/main/java/com/example/plandial/IconRecommendation.java:
```diff
@@ -0,0 +1,154 @@
+    private boolean getWordVectors(Context context){
+        // Word2Vec 모델 준비
+        try{
+            InputStream inputStream = context.getAssets().open("jsons/word_vectors.csv");
```
"jsons/word_vectors.csv" 는 상수로 선언해주세요

Mango-Juice replied:
변경 완료하였습니다. ([refactor : 폴더명 변경, 코드 간략화 및 상수 선언](https://github.com/ProjectFRIDAY/Plan-Dial/pull/4/commits/6054c425cedadc63f1f2c03ad5565168f4fec0da))

CaChiJ commented on app/src/main/java/com/example/plandial/IconRecommendation.java:
```diff
@@ -0,0 +1,154 @@
+
+            String json = new String(buffer, StandardCharsets.UTF_8);
+            JSONObject jsonObject = new JSONObject(json);
+            JSONArray jsonArray = jsonObject.getJSONArray("data");
```
"data" 는 상수로 선언해주세요

Mango-Juice replied:
변경 완료하였습니다. ([refactor : json 파일의 키값 상수 선언](https://github.com/ProjectFRIDAY/Plan-Dial/pull/4/commits/d5c662bafc41c57c0e7352f6a1151590cb324dba))

CaChiJ commented on app/src/main/java/com/example/plandial/IconRecommendation.java:
```diff
@@ -0,0 +1,154 @@
+            for(int i = 0; i < jsonArray.length(); i++)
+            {
+                JSONObject iconObject = jsonArray.getJSONObject(i);
+                sIconFileNames.add(iconObject.getString("icon") + ".png");
```
"icon"과 ".png" 모두 상수로 선언해주세요

Mango-Juice replied:
변경 완료하였습니다. ([refactor : 폴더명 변경, 코드 간략화 및 상수 선언](https://github.com/ProjectFRIDAY/Plan-Dial/pull/4/commits/6054c425cedadc63f1f2c03ad5565168f4fec0da), [refactor : json 파일의 키값 상수 선언](https://github.com/ProjectFRIDAY/Plan-Dial/pull/4/commits/d5c662bafc41c57c0e7352f6a1151590cb324dba))

CaChiJ commented on app/src/main/java/com/example/plandial/IconRecommendation.java:
```diff
@@ -0,0 +1,154 @@
+                sIconFileNames.add(iconObject.getString("icon") + ".png");
+
+                ArrayList<Double> vectorValues = new ArrayList<>();
+                JSONArray vectorArray = iconObject.getJSONArray("vector");
```
"vector"는 상수로 선언해주세요.

Mango-Juice replied:
변경 완료하였습니다. ([refactor : json 파일의 키값 상수 선언](https://github.com/ProjectFRIDAY/Plan-Dial/pull/4/commits/d5c662bafc41c57c0e7352f6a1151590cb324dba))

donghk0128 commented on app/src/main/java/com/example/plandial/IconRecommendation.java:
```diff
@@ -0,0 +1,154 @@
+        }
+    }
+
+    private boolean getWordVectors(Context context){
```
parameter와 중괄호 사이 space로 띄워야 합니다.

Mango-Juice replied:
넵 방급 수정했습니다!!

Mango-Juice committed:
style : 열린 중괄호 앞 띄어쓰기 적용

열린 중괄호 앞에 띄어쓰기를 적용했습니다.

CaChiJ commented on app/src/main/java/com/example/plandial/IconRecommendation.java:
```diff
@@ -0,0 +1,154 @@
+            return sIconFileNames.get(targetIndex);
+
+        } catch(Exception e){
+            return "unknown.png";
```
"unknown.png"는 상수로 선언해주세요

Mango-Juice replied:
변경 완료하였습니다. ([refactor : 폴더명 변경, 코드 간략화 및 상수 선언](https://github.com/ProjectFRIDAY/Plan-Dial/pull/4/commits/6054c425cedadc63f1f2c03ad5565168f4fec0da))

CaChiJ reviewed (commented):
코딩 스타일 관련한 리뷰입니다.

Mango-Juice committed:
style : 멤버 변수 앞 m 제거

멤버 변수 앞 m을 제거했습니다.

Mango-Juice committed:
refactor : 폴더명 변경, 코드 간략화 및 상수 선언

jsons 폴더의 이름을 datas로 변경하였습니다.
iconRecommendation.java 파일의 불필요한 코드를 줄이고 상수를 추가하였습니다.

Mango-Juice commented:
jsons 폴더에 json이 아닌 파일이 있는 것이 이상한 것 같아 datas로 이름 변경하였습니다.

Mango-Juice committed:
refactor : json 파일의 키값 상수 선언

json 파일의 키 값들을 상수로 선언하였습니다.
